platform/logger_api.h

```diff
@@ -0,0 +1,75 @@
+
+/** \addtogroup platform */
+/** @{*/
+/**
+ * \defgroup platform_log_api Logging API functions
+ * @{
+ */
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2013 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+#ifndef MBED_LOGGER_API_H
+#define MBED_LOGGER_API_H
```

Comment:
Not sure this file should be named mbed_logger_api.h. Usually *_api.h files are HAL files, otherwise aren't all header files api files?

Reply:
We have few api files in platform as well. Naming is not issue (suggestions welcome), I need two header files, one which will be added to "mbed.h" and other internal

Reply:
You could name it mbed_logger_internal.h. I think that's more clear.

Although you could just have all the macros in on big header file. IMO fewer files makes it easier to find things. Less jumping around. Also moving utilities to mbed_preprocessor.h would help reduce what you see in the file.

Reply:
`internal header file` that  reduces scope for internal things  sounds  good. That would answer also the question above about `#define LOG_ERR_CRITICAL        SEPARATOR "C" SEPARATOR`.

Reply:
Internal header files can reside in same directory?